prowler/providers/kubernetes/kubernetes_provider.py

```diff
@@ -295,12 +309,39 @@ def get_all_namespaces(self) -> list[str]:
             namespaces = [item.metadata.name for item in namespace_list.items]
             logger.info("All namespaces retrieved successfully.")
             return namespaces
+        except ApiException as api_error:
+            logger.critical(
+                f"ApiException[{api_error.__traceback__.tb_lineno}]: {api_error}"
+            )
+            raise KubernetesApiError(
+                original_exception=api_error, file=os.path.abspath(__file__)
+            )
+        except Timeout as timeout_error:
+            logger.critical(
+                f"Timeout[{timeout_error.__traceback__.tb_lineno}]: {timeout_error}"
+            )
+            raise KubernetesTimeoutError(
+                original_exception=timeout_error, file=os.path.abspath(__file__)
+            )
+        except ValueError as value_error:
+            logger.critical(
+                f"ValueError[{value_error.__traceback__.tb_lineno}]: {value_error}"
+            )
+            raise KubernetesValueError(
+                original_exception=value_error, file=os.path.abspath(__file__)
+            )
+        except TypeError as type_error:
+            logger.critical(
+                f"TypeError[{type_error.__traceback__.tb_lineno}]: {type_error}"
+            )
+            raise KubernetesTypeError(
+                original_exception=type_error, file=os.path.abspath(__file__)
+            )
         except Exception as error:
             logger.critical(
                 f"{error.__class__.__name__}[{error.__traceback__.tb_lineno}]: {error}"
             )
-            # TODO: add custom exception once we have the Kubernetes exceptions
-            raise error
+            raise KubernetesGetAllNamespacesError(original_exception=error)
```

Comment:
I'm thinking that maybe it is better to create a generic exception instead one per function, something like `KubernetesBaseException` with some code to wrap all the generic ones.

Reply:
Done